### ISSUE TITLE
Large delay in getting results from PySCF with the latest ElectronicStructureDriver class when a higher-order MO basis like cc-pVTZ is used

### ISSUE THREAD

**Issue by nirmal-aruvayi** [Qiskit/qiskit-nature#672]
When I tried to call PySCFDriver with a higher-order basis set (like ccpvdz), it's taking much time to run and return the QMolecule object. With the earlier version of PySCFDriver, where we could directly call PySCFDriver and feed molecule object, it was not taking much time with ccpvdz basis. In the latest version, where we use ElectronicStructureMoleculeDriver and set driver_type=ElectronicStructureDriverType.PYSCF is taking a bit longer even for the H2O molecule for ccpvdz basis set

**Comment by woodsp-ibm** [comment]
@nirmal-aruvayi  What version of Qiskit Nature are you working with? I will note that in the main branch of the repo here there have been several performance updates, such as #638 and #586 that affected the performance of creating an operator given a molecule, which are not yet released. Would you like to try installing Nature from source here and see if that fixes things for you. If you have not installed from source before there is further information here in the [installation instructions for Nature](https://qiskit.org/documentation/nature/getting_started.html#installation).

**Comment by mrossinek** [comment]
The more important improvement which has recently been merged is actually #621 
See also the example for `6-31g*` I provide in that PR.

Starting with Qiskit Nature 0.4.0, this improvement will be enabled by default, so once that is released, you should be able to benefit from this immediately. If, however, you decide to attempt the installation from source, you will also need to add the following to your test code:
```python
from qiskit_nature.settings import settings

settings.optimize_einsum = True
```

**Comment by nirmal-aruvayi** [comment]
Thanks Max and Wood.

My sincere apologies for this very late reply as I was caught up with
something else.

I'm running all my codes on IBM Quantum Lab and I see that the Qiskit
Nature version being installed is 0.3.2. So, my query is has Qiskit-Nature
0.4.0 released yet, if so, can I pip install qiskit-ature == 0.4.0 in IBM
Quantum Lab and try the latest version of the code? Just a few days back, I
got an error of missing the PySCFDriver while running the PySCFDriver
statement. I then pip installed qiskit-nature[pyscf] and started running
again

Also, I have been importing PySCFDriver as: from qiskit_nature.drivers
import PySCFDriver
This import statement has given the Deprecation Warning. So, if I import
from qiskit_nature.drivers.second_quantization import PySCFDriver, will it
work fine as done by Max in the example code? Or shall I start using the
latest ElectronicStructureDriver class and then calling the PySCFDriver as
an argument?

**Comment by woodsp-ibm** [comment]
Hi, 0.4.0 has not yet been released but should do so very soon now.

As you have seen the former drivers were deprecated and moved/refactored to second_quantization. If by Max's example you mean the code in the performance issue, yes that would work in 0.4.0. You can also look at the [Qiskit Nature tutorials](https://qiskit.org/documentation/nature/tutorials/) and/or the readme example for [Creating Your First Chemistry Programming Experiment in Qiskit](https://github.com/Qiskit/qiskit-nature#creating-your-first-chemistry-programming-experiment-in-qiskit). All use the newer way. In 0.4.0 the former drivers are removed so only the newer code is now possible.

When you say use the `ElectronicStructureDriver` did you mean `ElectronicStructureMoleculeDriver`. If so that's a class which takes `Molecule` and represents a common interface that can be used across drivers - it will auto-select an installed driver or you can specify which one it should use. You can thing of this somewhat as the lowest common denominator in that it will work across all drivers. If you are already using PySCF directly then you can continue to do so directly with the second_quantization driver replacement.

**Comment by woodsp-ibm** [comment]
@nirmal-aruvayi The release with the fix is now available - 0.4.1 (we did a bug fix soon after releasing 0.4.0 when a compatibility issue arose). Let us know if this fixes the issue you reported (it should) and if so please close this issue as resolved, thx.
